internal/image/authorization_test.go

```diff
@@ -18,50 +18,62 @@ package image
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"testing"
 
 	ecc "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
 	"github.com/stretchr/testify/assert"
 )
 
-func mockFetchECSource(ctx context.Context, resource string) (*ecc.EnterpriseContractPolicy, error) {
+func mockFetchECSource(ctx context.Context, resource string) (*ecc.EnterpriseContractPolicySpec, error) {
 	description := "very descriptive"
-	return &ecc.EnterpriseContractPolicy{
-		Spec: ecc.EnterpriseContractPolicySpec{
-			Description: &description,
-		},
+	return &ecc.EnterpriseContractPolicySpec{
+		Description: &description,
 	}, nil
 }
 
+func mockPolicyConfigurationString() string {
+	description := "very descriptive"
+	config := &ecc.EnterpriseContractPolicySpec{
+		Description: &description,
+	}
+	configJson, _ := json.Marshal(config)
+	return string(configJson)
+}
+
 func Test_NewK8sSource(t *testing.T) {
 	wanted := &K8sSource{
-		namespace:   "enterprise-contract",
-		server:      "k8s-server",
-		resource:    "ecp/resource",
-		fetchSource: mockFetchECSource,
+		policyConfiguration: mockPolicyConfigurationString(),
+		fetchSource:         mockFetchECSource,
 	}
 
-	source, err := NewK8sSource("k8s-server", "enterprise-contract", "ecp/resource")
+	source, err := NewK8sSource(mockPolicyConfigurationString())
 	assert.ObjectsAreEqualValues(wanted, source)
 	assert.NoError(t, err)
 }
 
 func Test_GeKk8sSignOff(t *testing.T) {
 	input := &k8sResource{
-		RepoUrl: "my-git-repo",
-		Sha:     "1234",
-		Author:  "ec@redhat.com",
+		Components: []ecc.AuthorizedComponent{
+			{
+				ChangeID:   "1234",
+				Repository: "my-git-repo",
+				Authorizer: "ec@redhat.com",
+			},
+		},
 	}
-	expected := &authorizationSignature{
-		RepoUrl:     "my-git-repo",
-		Commit:      "1234",
-		Authorizers: []string{"ec@redhat.com"},
+	expected := []authorizationSignature{
+		{
+			RepoUrl:     "my-git-repo",
+			Commit:      "1234",
+			Authorizers: []string{"ec@redhat.com"},
+		},
 	}
 
 	signOff, err := input.GetSignOff()
 	assert.NoError(t, err)
-	assert.Equal(t, expected, signOff)
+	assert.ObjectsAreEqualValues(expected, signOff)
```

Comment:
Oh! This actually doesn't cause a test failure. It just returns a boolean. [ObjectsAreEqualValues](https://pkg.go.dev/github.com/stretchr/testify/assert#ObjectsAreEqualValues). The tell-tale is that it doesn't take a `TestingT` parameter like `Equal` does. So it does not have a way to mark the test as failed. We should fix that here, and in the other places in this file.

Reply:
Good catch! I simplified the test and removed that function call.